File: backend/src/TheBlueSky.Flights/Models/Flight.cs
using System.ComponentModel.DataAnnotations;
using TheBlueSky.Flights.Enums;

namespace TheBlueSky.Flights.Models
{
    public class Flight
    {
        public int FlightId { get; set; }

        [Required]
        public int FlightScheduleId { get; set; }

        [Required]
        public DateOnly FlightDate { get; set; }

        [Required]
        public DateTimeOffset DepartureDateTime { get; set; }

        [Required]
        public DateTimeOffset ArrivalDateTime { get; set; }

        [Required]
        public FlightStatus FlightStatus { get; set; } = FlightStatus.Scheduled;


        public int AvailableSeats { get; set; }

        [Required]
        public DateTime LastUpdated { get; set; } = DateTime.UtcNow;

        public FlightSchedule Schedule { get; set; } = null!;
        public ICollection<FlightSeatStatus> SeatStatuses { get; set; } = new List<FlightSeatStatus>();

    }
}

File: frontend/TheBlueSky.App/src/types/flight.ts
import type { Aircraft } from "./aircraft";
import type { AircraftSeat } from "./aircraftSeat";
import type { FlightSchedule } from "./flightSchedule";
import type { flightSeatStatus } from "./flightSeatStatus";

export const flightStatuses = [
  'Scheduled',
  'Boarding',
  'Departed',
  'Arrived',
  'Delayed',
  'Cancelled',
  'Diverted',
] as const;

export type FlightStatus = (typeof flightStatuses)[number];

export type Flight = {
  flightId: number;
  flightScheduleId: number;
  flightDate: string; // YYYY-MM-DD
  departureDateTime: string;
  arrivalDateTime: string;
  flightStatus: FlightStatus;
  baseFare: number,
  availableSeats: number;
};

export type GetFlight = Flight &{
  schedule: FlightSchedule & {
    aircraft: Aircraft
  },
  seatStatuses: flightSeatStatus & {
    aircraftSeat: AircraftSeat
  }
}